bcs/deserializer.go

```diff
@@ -215,19 +215,28 @@ func (des *Deserializer) Struct(v Unmarshaler) {
 }
 
 // DeserializeSequence deserializes an Unmarshaler implementation array
+// TODO: can T be of type Unmarshaler?
 func DeserializeSequence[T any](des *Deserializer) []T {
+	return DeserializeSequenceWithFunction(des, func(des *Deserializer, out *T) {
+		mv, ok := any(out).(Unmarshaler)
+		if ok {
+			mv.UnmarshalBCS(des)
+		}
```

Comment:
this lost the else case of setting the error. if the typecast fails this will now silently fail